Source/Csla.AspNetCore/Blazor/ApplicationContextManagerBlazor.cs

```diff
@@ -187,18 +185,20 @@ private void SetHostPrincipal(Task<AuthenticationState> task)
     /// </summary>
     public ContextDictionary GetLocalContext()
     {
-      if (LocalContext is null)
+      ContextDictionary localContext;
+      var sessionManager = ApplicationContext.GetRequiredService<ISessionManager>();
+      var session = sessionManager.GetSession();
+      session.TryGetValue("localContext", out var result);
```

Comment:
What about compacting this whole method to:
```csharp
if(!session.TryGetValue("localContext", out var storedContext) || storedContext is not ContextDictionary context) {
	context = [];
	SetLocalContext(context);
}

return context;
```
I don't know if a current TFM is impeding on that?

Reply:
Does this increase or decrease readability? I argue that it decreases readability.

Reply:
@rockfordlhotka Just to answer since this is a subjective thing here.
I think it increases readability but at a second thought I would invert the if to:
```csharp
if(session.TryGetValue("localContext", out var storedContext) && storedContext is ContextDictionary context) {
  return ctx;
}

ContextDictionary context = [];
SetLocalContext(context);
return context;
```

I personally don't like/use the "declare variables at the beginning of a method regardless where it's used". So I try to declare it as close as possible to it's first usage. This method is quite short so it's not that far but it still took me two or three looks to see where the variable was declared and initialized.